conceptnet5/vectors/evaluation/analogy.py

```diff
@@ -136,7 +137,42 @@ def read_turk_ranks_semeval2012(subset, subclass):
         return sorted(gold_ranks)
 
 
-def analogy_func(wrap, a1, b1, a2, weight_direct=2/3, weight_transpose=1/3):
+def read_bats(category):
+    """
+    Read a bats dataset pairs for a specific category. Turn them into questions.
+    """
+    filename = 'bats/{}.txt'.format(category)
+    pairs = []
+    with open(get_support_data_filename(filename)) as file:
+        for line in file:
+            if '\t' in line:
+                left, right = line.lower().split('\t')
+            else:
+                left, right = line.lower().split()
+            right = right.strip()
+            if '/' in right:
+                right = [i.strip() for i in right.split('/')]
+            else:
+                right = [i.strip() for i in right.split(',')]
+            pairs.append([left, right])
+
+    quads = []
+    for i in range(len(pairs)):
+        a_pair = pairs[i]
+        a_pair[1] = a_pair[1][0]  # select only the first element if b1 is a list of synonyms
```

Comment:
What if b1 isn't a list of synonyms? Is this replacing it by its first character?

Reply:
This was an incorrect comment. b1 is always a list, therefore it's safe to do `a_pair[1][0]`